src/library/scala/concurrent/Future.scala

```diff
@@ -677,8 +676,15 @@ object Future {
    */
   def firstCompletedOf[T](futures: TraversableOnce[Future[T]])(implicit executor: ExecutionContext): Future[T] = {
     val p = Promise[T]()
-    val completeFirst: Try[T] => Unit = p tryComplete _
-    futures foreach { _ onComplete completeFirst }
+    val firstCompleteHandler = new AtomicReference[Promise[T]](p) with (Try[T] => Unit) {
+      override def apply(v1: Try[T]): Unit = {
```

Comment:
```override def apply(v1: Try[T]): Unit = getAndSet(null) match {
  case null =>  @()
  case some => some tryComplete v1
}```

Reply:
matter of style I guess, will adopt to the local style :sunglasses:

Reply:
Awesome, thank you :+1: :)